### PR TITLE
Id string comparison

### PR DESCRIPTION
**Pull request by idimopoulos**
As the branch name declares, this PR is to support the operators '<', '>', '<=', '>=' for the id field.

*Disclaimer:*
The process of comparing and sorting per id was already supported. What is added here is the option to filter by id greater(or equal)/less(or equal) than some string. This process is very slow and not recommended if it can be avoided. In a very basic test I did the following:
```
SELECT ?a ?b ?c
WHERE {
  ?a ?b ?c
}
LIMIT 100
```
Process time: 86 ms.

```
SELECT ?a ?b ?c
WHERE {
  ?a ?b ?c .
  FILTER (STR(?a) < STR(<http://a))
}
LIMIT 100
```
Process time: 646 ms.

Still, I also added tests for all that matters.

### CODE REVIEW

**Comment by pfrenssen** [comment]
The longer process time is a pity, but I guess it is normal that a query with a condition takes longer than a straight query without conditions. I don't know enough about Sparql / Virtuoso to judge whether this is the most optimal way to do this, but I can see from the tests that it is working.

**Comment by idimopoulos** [comment]
I don't think it's the condition that delays it that much. I think it's the conversion that does it. Unfortunately, you cannot use  a condition like `?entity_iri_1 < ?entity_iri_2` where iri is a `<http://example.com>` url. It needs to be converted in order to use the `<, <=, >, >=` operators.

File: src/Entity/Query/Sparql/SparqlCondition.php [hunk 1]
```diff
@@ -767,10 +786,18 @@ protected function escapeValue($field, $value, $column = NULL, $lang = NULL) {
...
+
     if ($field === $this->idKey) {
-      return SparqlArg::toResourceUris($value);
+      // If the field name is the id, and the operators are not among the '>',
+      // '<', '>=', '<=', then the value has already been converted in an array
+      // so escape it.
+      if (is_array($value)) {
+        return SparqlArg::toResourceUris($value);
+      }
+      // Convert the value to sting in order to perform the rest of comparisons.
```
Comment (pfrenssen) [inline src/Entity/Query/Sparql/SparqlCondition.php]: sting? :dagger:

**Comment by pfrenssen** [comment]
Looking good! Thanks!

**Pull request merged**